src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -61,72 +59,85 @@ public Boolean upsertNode(Node node) throws Exception {
-
-                Statement statement = new Statement(queryNodeExistWithReverseEdge, parameters);
-                StatementResult result = transaction.run(statement);
-                int recordSize = result.list().size();
-                result.consume();
-                if (recordSize != 0) {
-                    throw new GraphException(ErrorCode.NODES_WITH_REVERSE_EDGE_EXISTS.name(),
-                            "Nodes exists with reverse edge, new relation cannot be created!");
-                }
+            String queryNodeExistWithReverseEdge = "MATCH (n:" + label + ")<-[r:connect]-(n1:" +
+                    label + ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "RETURN n,n1";
 
+            Statement statement = new Statement(queryNodeExistWithReverseEdge, parameters);
+            StatementResult result = transaction.run(statement);
+            int recordSize = result.list().size();
+            result.consume();
+            if (recordSize != 0) {
+                if (logger.isDebugEnabled())
+                    logger.debug(nodeFrom.getId(), nodeTo.getId());
+                updateRelationshipBetweenTwoNodes(nodeFrom, nodeTo, statement, result, transaction, recordSize, relationProperties);
+                transaction.commitAsync().toCompletableFuture().get();
+                logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
+                        nodeFrom.getId());
+            } else {
                 String query = "MATCH (n:" + label + ")-[r:connect]->(n1:" + label +
-                        ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "SET r" + " += " +
-                        "$props " + "RETURN n,n1";
+                        ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "RETURN n,n1";
 
                 statement = new Statement(query, parameters);
                 result = transaction.run(statement);
                 recordSize = result.list().size();
                 result.consume();
                 if (recordSize == 0) { // nodes relation doesn't exists
-                    createRelationshipBetweenTwoNodes(nodeFrom, nodeTo, transaction, parameters);
+                    isUpserted = createRelationshipBetweenTwoNodes(nodeFrom, nodeTo, transaction, parameters);
+                } else {
+                    if (logger.isDebugEnabled())
+                        logger.debug(nodeTo.getId(), nodeFrom.getId());
+                    isUpserted = updateRelationshipBetweenTwoNodes(nodeTo, nodeFrom, statement, result, transaction, recordSize, relationProperties);
                 }
                 transaction.commitAsync().toCompletableFuture().get();
-                logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
+                logger.info("user relation with toUUID {} and fromUUID {} added successfully ", nodeTo.getId(),
```

Comment:
There is already an Info log printed inside both create and update relation... this is not required... if you need more info, use the Debug statement instead of info logs